UI.cpp

```diff
@@ -0,0 +1,296 @@
+            std::cout << std::endl << std::endl << "----------- Login -----------" << std::endl;
+            std::string name = user_name();
+            std::string pass = password();
+            system("cls");
+            account = this->logIn(name, pass,account);
+            if (account.get_name()!="") {
+                return account;
+            }
+            system("pause");
+            system("cls");
+        }
+        else
+            if (opt == 0) {
+                system("cls");
+                std::cout << std::endl << "----------- Sing in -----------" << std::endl;
+                std::string name = user_name();
+                std::string pass = password();
+                system("cls");
+                id = id + 1;
+                this->signIn(name, pass,id);
+                system("pause");
+                system("cls");
+            }
+    } while (true);
+}
+
+void UI::admin_menu() {
+    cout << "--------- Admin Menu ---------\n";
+    cout << "0.Exit\n";
+    cout << "1.Search users by id\n";
+    cout << "2.Post\n";
+    cout << "3.Delete user\n";
+    cout << "4.Delete post\n";
+    cout << "5.Update user type\n";
+    cout << "6.Show all\n";
+}
+
+void UI::user_menu() {
+    cout << "--------- User Menu ---------\n";
+    cout << "0.Exit\n";
+    cout << "1.Search users by id\n";
+    cout << "2.Post\n";
+    cout << "3.Delete post\n";
+    cout << "4.Show all\n";
+}
+
+void UI::printPosts(vector<Post> arr) {
+
+    int i = 0;
+    for (i = 0; i<int(arr.size()); i++) {
+        cout << arr[i]<<"\n";
+    }
+}
+
+void UI::findUser() {
+    string name;
+    cout << "Name: ";
+    cin >> name;
+    Account account = service.find_user(name);
+    if (account.get_name() == name)
+        printPosts(service.find_user_post(account));
+    else
+        cout << "User does not exist\n";
+}
+void UI::findUser(string name) {
+    
+    Account account = service.find_user(name);
+    if (account.get_name() == name)
+        printPosts(service.find_user_post(account));
+    else
+        cout << "User does not exist\n";
+}
+
+void UI::post(Account account) {
+    string title, posting;
+    cout << "Title: \n";
+    cin >> title;
+    cout << "Post: \n";
+    cin >> posting;
+    this->service.toPost(title, posting,account);
+}
+
+void UI::display_accounts() {
+    vector<Account>arr = service.display_users();
+    for (int i = 0; i<int(arr.size()); i++) {
+        cout << arr[i];
+    }
+}
+
+void UI::updateUser() {
+
+    this->display_accounts();
+    string name;
+    cout << "User: ";
+    cin >> name;
+    Account account = service.find_user(name);
+    if (account.get_name() == name) {
+        service.updateUser(account);
+    }
+    else
+        cout << "User does not exist!";
+}
+
+void UI::run_admin(Account account) {
+    int opt;
+    do {
+        this->admin_menu();
+        cin >> opt;
+        if (opt == 1) {
+            this->findUser();
+            system("pause");
+            system("cls");
```

Comment:
For each of these options you will execute something.
I think you could use a `switch` statement to make this easier, because you won't have 2 options that will call the same method. Also, you can move everything from a call, for example 
```c++
            this->findUser();
            system("pause");
            system("cls");
```
could be in a new private method called `findUserUI` or something like this. It could be easier and helpfull in the future.